IntelliTect.Multitool/IPaths.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace IntelliTect.Multitool;
+/// <summary>
+/// Provides normalized paths.
+/// </summary>
+public interface IPaths
+{
+    /// <summary>
+    /// Returns the repository root directory name.
+    /// </summary>
+    /// <param name="CurrentProjectFolderName">The highest level directory name of the current project within the repository.</param>
+    /// <returns></returns>
+    public static string GetDefaultRepoRoot(string CurrentProjectFolderName)
+    {
+        DirectoryInfo? currentDirectory = new(Directory.GetCurrentDirectory());
+
+        while (currentDirectory is not null)
+        {
+            if (currentDirectory.FullName.ToLowerInvariant().EndsWith(CurrentProjectFolderName.ToLowerInvariant(), StringComparison.InvariantCulture))
```

Comment:
Generally you want to avoid the `ToLower*()` methods whenever possible as this is creating copies of strings. A better solution is to user the EndsWith overrides to do a case insensitive comparison.
```suggestion
            if (currentDirectory.FullName.EndsWith(CurrentProjectFolderName, StringComparison.InvariantCultureIgnoreCase))
```